HIP/hip-1056.md

```diff
@@ -0,0 +1,1952 @@
+In this way any consumer of a block would have the complete state by applying the diffs onto the initial state.
+This shares the downside of requiring additional temporary storage, and also requires the creation of a "difference" form for every
+entity potentially stored in state, and could, paradoxically, _increase_ the total size of blocks with very many changes to small entities.
+
+### Including exchange_rate in the TransactionResult
+In the past the exchange_rate details were in the transaction record. However, this information doesn't change often and is based on
+the value of a system file. Changes to this file are externalized to the MN and thus the current exchange rate value is always available
+via a MN (or future BN). Thus the exchange rate is readily available from the MN when using the transaction timestamp as input.
+This allowed us to removing it and reduce unnecessary data from the block stream.
+
+
+## Open Issues
+Open Issues captured here and in https://github.com/hashgraph/hedera-block-node/issues/37
+
+- [ ] Q: Are the current block sizes acceptable to not need to adopt an entity state changes splitting strategy?
+- [ ] Q: Is it clear which block contains a block proof and state hash regarding that block?
+    - Currently a block will contain it's own proof. The state hash a block contains however is the hash at the begining of the block/end
+    of the previous block. The state hash representing the changes to state made by a given block are found in the next block. 
+    State proofs for block N thus require the state following block N, and the block proof from block N+1.
+- [ ] Q: There are some cases where multiple changes made to a singleton state type over the course of a round will be consolidated into a single change set for that singleton type for the round instead. How do those fit into the stream and block merkle tree? Do they need to be placed somewhere specific both in the stream and in the tree?
+- [ ] Q: Should TransferList be included in the TransactionResult object or not.  It seems closer to an input and output.
+    - Couldn't a MN that parses the block stream take changes to the transferList in the TransactionResult? 
+    There are considerations for emergent transfer logic that isn’t in the transaction body which may prevent this.
+- [ ] Q: Is the current algorithm for Block hash calculation accurate based on latest services explorations?
+- [ ] Q: Should we include a string error message in the block stream to give improved DexExp when troubleshooting transaction failures?
+- [ ] Q: Are redaction and errata and how FilteredBlockItem can be used to achieve them adequately described?
+- [ ] Q: Can we change EthereumOutput to EVMOutput? it seems more accurate to cover EVM transactions
+    - An argument against this is that Outputs match to the transaction input. Since there's a ContractCall, ContractCreate and
+    EThereumTransaction it would be inconsistent.
```

Comment:
```suggestion
- [ ] Q: Is the current compression approach sufficient to maintain manageable block size and
      avoid adopting an entity state changes splitting strategy?
- [ ] Q: Are there additional protocol buffer changes needed to reduce the stream overhead
      for Events, Transactions, or other frequent entities.
- [ ] Q: Is it clear which block contains a block proof and state hash regarding that block?
   - A block will contain it's own proof. The state hash a block contains however is the
     hash at the begining of the block/end of the previous block. The state hash representing the
     changes to state made by a given block are found in the next block. State proofs after block N
     thus require the state at the end of block N, and the block proof from block N+1.
- [ ] Q: There are some cases where multiple changes made to a singleton state type over the course
      of a round will be consolidated into a single change set for that singleton type for the
      round instead. How do those fit into the stream and block merkle tree? Do they need to be
      placed somewhere specific both in the stream and in the tree?
   - Actually, it's possible these must be reported multiple times instead of holding them
     to a single later update. The reason being that out-of-order state changes, or skipped
     state changes could produce a different total order of updates, and change the state hash.<br/>
     This needs to be verified with the services, platform, and platform-data experts.
- [ ] Q: Should TransferList be included in the TransactionResult object or not. It contains
      redundant data, but makes mirror node processing much simpler.
   - Could a MN that parses the block stream reconstruct the transfer list from state changes?<br/>
     There are considerations for emergent transfer logic that isn’t in the transaction body which
     may prevent this.
- [ ] Q: Is the current algorithm for Block hash calculation accurate based on latest services
      explorations?
- [ ] Q: Should we include a string error message in the block stream to give improved DexExp when
      troubleshooting transaction failures?
   - Should we, alternatively, offer a detail error _number_, defined on a per-result-code basis,
     and published as a set of  enumerations? This numeric option would greatly reduce extra data
     in the block stream and avoid unclear or "bit-rotted" error strings. 
- [ ] Q: Are redaction and errata and how FilteredBlockItem can be used to achieve them adequately
      described?
- [ ] Q: Can we change EthereumOutput to EVMOutput? it seems more accurate to cover EVM transactions
   - An argument against this is that Outputs match to the transaction input. Since there's a
     ContractCall, ContractCreate and EthereumTransaction it would be inconsistent.
```